phpBB/phpbb/lock/db.php

```diff
@@ -110,7 +110,18 @@ public function acquire()
 		// process we failed to acquire the lock.
 		$this->locked = $this->config->set_atomic($this->config_name, $lock_value, $this->unique_id, false);
 
-		return $this->locked;
+		if ($this->locked == true)
```

Comment:
the logic here is wrong, and nothing could be returned with this. combine these conditions, and again the else is not needed.